mink/tasks/equality_constraint_task.py

```diff
@@ -0,0 +1,186 @@
+                f"{self.__class__.__name__} no equality constraints found in this "
+                "model."
+            )
+        self._eq_ids = np.array(eq_ids)
+        self._eq_types = model.eq_type[self._eq_ids]
+        self._neq_total = len(self._eq_ids)  # Total number of equality constraints.
+
+        cost_dim = _get_cost_dim(self._eq_types)
+        super().__init__(cost=np.zeros((cost_dim,)), gain=gain, lm_damping=lm_damping)
+        self.set_cost(cost)
+
+    def set_cost(self, cost: npt.ArrayLike) -> None:
+        cost = np.atleast_1d(cost)
+        if cost.ndim != 1 or cost.shape[0] not in (1, self._neq_total):
+            raise TaskDefinitionError(
+                f"{self.__class__.__name__} cost must be a vector of shape (1,) "
+                f"or ({self._neq_total},). Got {cost.shape}."
+            )
+        if not np.all(cost >= 0.0):
+            raise TaskDefinitionError(f"{self.__class__.__name__} cost must be >= 0")
+
+        # For scalar cost, broadcast to all elements.
+        if cost.shape[0] == 1:
+            self.cost[:] = cost[0]
+            return
+
+        # For vector cost, repeat each element according to its constraint dimension.
+        repeats = [_get_constraint_dim(eq_type) for eq_type in self._eq_types]
+        self.cost[:] = np.repeat(cost, repeats)
+
+    def _update_constraint_info(self, configuration: Configuration) -> None:
+        self._mask = _get_equality_constraint_indices(configuration.data, self._eq_ids)
+        self._neq_active = np.sum(self._mask)
+
+    def compute_error(self, configuration: Configuration) -> np.ndarray:
+        """Compute the equality constraint task error.
+
+        Args:
+            configuration: Robot configuration :math:`q`.
+
+        Returns:
+            Equality constraint task error vector :math:`e(q)`. The shape of the
+            error vector is ``(neq_active * constraint_dim,)``, where ``neq_active``
+            is the number of active equality constraints, and ``constraint_dim``
+            depends on the type of equality constraint.
+        """
+        self._update_constraint_info(configuration)
+        return configuration.data.efc_pos[self._mask]
+
+    def compute_jacobian(self, configuration: Configuration) -> np.ndarray:
+        """Compute the task Jacobian at a given configuration.
+
+        Args:
+            configuration: Robot configuration :math:`q`.
+
+        Returns:
+            Equality constraint task jacobian :math:`J(q)`. The shape of the Jacobian
+            is ``(neq_active * constraint_dim, nv)``, where ``neq_active`` is the
+            number of active equality constraints, ``constraint_dim`` depends on the
+            type of equality constraint, and ``nv`` is the dimension of the tangent
+            space.
+        """
+        self._update_constraint_info(configuration)
+        data = configuration.data
+        J_efc = data.efc_J.reshape((data.nefc, configuration.model.nv))
```

Comment:
So I'm guessing somewhere higher up you already error if mj_isSparse()?

Reply:
Fixed now, nice catch!